packages/library-base/src/commonMain/kotlin/io/realm/Configuration.kt

```diff
@@ -21,6 +21,19 @@ import io.realm.log.RealmLogger
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlin.reflect.KClass
 
+/**
+ * This interface is used to determine if a Realm file should be compacted the first time the file
+ * is opened and before the instance is returned.
+ *
+ * Note that compacting a file can take a while, so compacting should generally only be done when
+ * opening a realm file on a background thread.
+ *
+ * - `totalBytes` is the total file size (data + free space).
+ * - `usedBytes` is the total bytes used by data in the file.
+ * - `true` should be returned if an attempt to compact the file should be made. `false` if not.
+ */
+typealias CompactOnLaunchCallback = (totalBytes: Long, usedBytes: Long) -> Boolean
```

Comment:
I would make this an explicit named (functional?) interface. Don't think that the required indirection from the underlying lambda passed to interop to the user supplied callback is worth optimizing away. 

But I guess it would make sense to have a typemap in `cinterop` for the lambda.

Reply:
Ah yes, forgot that you could explicitly mark them as functional interfaces..that is probably the best of both worlds 👍